src/worldcereal/openeo/feature_extractor.py

```diff
@@ -0,0 +1,115 @@
+"""Feature computer GFMAP compatible to compute Presto embeddings."""
+
+import xarray as xr
+from openeo_gfmap.features.feature_extractor import PatchFeatureExtractor
+
+class PrestoFeatureExtractor(PatchFeatureExtractor):
+    """Feature extractor to use Presto model to compute embeddings.
+    This will generate a datacube with 128 bands, each band representing a
+    feature from the Presto model.
+    """
+
+    import functools
+    from pathlib import Path
+    from typing import Tuple
+
+    PRESTO_PATH = "https://artifactory.vgt.vito.be/artifactory/auxdata-public/worldcereal-minimal-inference/presto.pt"  # NOQA
+    BASE_URL = "https://s3.waw3-1.cloudferro.com/swift/v1/project_dependencies"  # NOQA
+    DEPENDENCY_NAME = "wc_presto_onnx_dependencies.zip"
+
+    GFMAP_BAND_MAPPING = {
+        "S2-L2A-B02": "B02",
+        "S2-L2A-B03": "B03",
+        "S2-L2A-B04": "B04",
+        "S2-L2A-B05": "B05",
+        "S2-L2A-B06": "B06",
+        "S2-L2A-B07": "B07",
+        "S2-L2A-B08": "B08",
+        "S2-L2A-B8A": "B8A",
+        "S2-L2A-B11": "B11",
+        "S2-L2A-B12": "B12",
+        "S1-SIGMA0-VH": "VH",
+        "S1-SIGMA0-VV": "VV",
+        "COP-DEM": "DEM",
+        "A5-tmean": "temperature-mean",
+        "A5-precip": "precipitation-flux",
+    }
+
+    def __init__(self):
+        """
+        Initializes the PrestoFeatureExtractor object, starting a logger.
+        """
+        import logging
+
+        logging.basicConfig(level=logging.INFO)
+        self.logger = logging.getLogger(PrestoFeatureExtractor.__name__)
+
+    @classmethod
+    @functools.lru_cache(maxsize=6)
+    def extract_dependencies(cls, base_url: str, dependency_name: str):
+        """Extract the dependencies from the given URL. Unpacking a zip
+        file in the current working directory.
+        """
+        import shutil
+        import urllib.request
+        from pathlib import Path
+
+        # Generate absolute path for the dependencies folder
+        dependencies_dir = Path.cwd() / "dependencies"
+
+        # Create the directory if it doesn't exist
+        dependencies_dir.mkdir(exist_ok=True, parents=True)
+
+        # Download and extract the model file
+        modelfile_url = f"{base_url}/{dependency_name}"
+        modelfile, _ = urllib.request.urlretrieve(
+            modelfile_url, filename=dependencies_dir / Path(modelfile_url).name
+        )
+        shutil.unpack_archive(modelfile, extract_dir=dependencies_dir)
+
+        # Add the model directory to system path if it's not already there
+        abs_path = str(
+            dependencies_dir / Path(modelfile_url).name.split(".zip")[0]
+        )  # NOQA
+
+        return abs_path
+
+    def output_labels(self) -> list:
```

Comment:
Have you tested if these are in fact attached to the dataset? I've seen in the past that band names altered in the UDF are not picked up by OpenEO unless specifically using an OpenEO rename.

Reply:
That's the idea of it. The Feature Computer is initialized first locally in order to call this `output_labels` function, which then will be added on the OpenEO computing graph